OsmAnd-java/src/main/java/net/osmand/search/core/SearchPhrase.java

```diff
@@ -318,6 +318,16 @@ private void calcMainUnknownWordToSearch() {
 			}
 		}
 	}
+	
+	private boolean hasMoreOneLetter(String s) {
+		int count = 0;
+		for (int i = 0; i < s.length(); i++) {
+			if (Character.isAlphabetic(s.charAt(i))) {
```

Comment:
dont' check is alphabetic to slow, check it's not a digit or space (for C++ version only '0'-'9' and ' ', '\t')